### DISCUSSION TITLE
upgrading from 1.26 to 1.28.x

### DISCUSSION THREAD

**Discussion by fontexD** [dani-garcia/vaultwarden#3538] [Q&A]
### Subject of the issue
upgrading from 1.26 to 1.28 causes db read arrors with ciphers 

### Deployment environment

running in k8s 1.26

* MySQL/MariaDB or PostgreSQL version: 

using mysql latest

### Expected behaviour
acceess to my data, i can see the orgs but no data in them since it cant read them i tried running the describe alter db collaction etc from the guide of mysql but didnt really fix it

### Troubleshooting data
[2023-05-24 11:17:22.562][panic][ERROR] thread 'rocket-worker-thread' panicked at 'Error loading ciphers: DatabaseError(Unknown, "Illegal mix of collations (utf8mb4_0900_ai_ci,IMPLICIT) and (utf8mb4_unicode_ci,IMPLICIT) for operation '='")': src/db/models/cipher.rs:633

**Comment by BlackDex** [comment]
See: https://github.com/dani-garcia/vaultwarden/wiki/Using-the-MariaDB-%28MySQL%29-Backend#foreign-key-errors-collation-and-charset

That should really fix all the issue. Just make sure all collation and charsets are correct.

**Reply by BlackDex** [reply]
Run the following two queries to see if all is correct.
Change `"vaultwarden"` to the correct database name for your deployment.

```sql
--- Check the defaults of the main database
SELECT DEFAULT_CHARACTER_SET_NAME, DEFAULT_COLLATION_NAME FROM information_schema.SCHEMATA WHERE SCHEMA_NAME = "vaultwarden";
--- Check the defaults of all the tables.
SELECT CHARACTER_SET_NAME, COLLATION_NAME FROM information_schema.`COLUMNS` WHERE TABLE_SCHEMA = "vaultwarden" AND CHARACTER_SET_NAME IS NOT NULL;
```

P.S. I just added this also to the wiki page for future reference :).

**Reply by fontexD** [reply]
https://ibb.co/XZXYbXL

**Reply by BlackDex** [reply]
That seems to be only the first query, not the second, and i think there lays the issues.

**Reply by fontexD** [reply]
https://ibb.co/1Q78gBC

**Reply by BlackDex** [reply]
That shows wrong collations, so those should be fixed, else you will keep issues.

**Reply by fontexD** [reply]
how do i fix these ? i tried with the described commands in the wiki

**Reply by BlackDex** [reply]
The commands on the wiki should work just fine. Just make sure you follow the instructions.
That is the only way to solve this. Or with some DB editor you can fix it manually, like DBeaver for example, but others will probably also work just fine.
